platform-dom/src/main/java/com/softicar/platform/dom/elements/time/day/DomDayInput.java

```diff
@@ -49,11 +53,40 @@ public void disable() {
 		dayButton.unlistenToEvent(DomEventType.CLICK);
 	}
 
+	/**
+	 * Parses the value text into a {@link Day}.
+	 * <p>
+	 * If the value text is empty or blank, an empty {@link Optional} is
+	 * returned. Otherwise, the value text is parsed into a {@link Day}. If
+	 * parsing failed, an exception is thrown.
+	 *
+	 * @return the optionally entered {@link Day}
+	 * @throws IllegalDateSpecificationException
+	 *             if the non-blank value text could not be parsed
+	 */
+	public Optional<Day> getDayOrThrowIfInvalid() {
```

Comment:
Again, we could consider removal of the `IfInvalid` suffix.

Applies to various places in the code below as well.

Reply:
- Well here I disagree, because the `OrThrow` could also mean, throw if invalid or empty.
- Actually, in the long run we should wipe a lot of methods on many inputs and renamed everything, so that there is only one method called `Optional<Day> getValue()` with the exact behavior of `getDayOrThrowIfInvalid()`

Reply:
For example, `IEmfInput.getValueAsOptional` does it right, although the `AsOptional` suffix could be removed.

Reply:
Maybe name like `parseValue` would be even better to indicate that there is some computation going on that could fail. ;) We should probably create some story in the backlog. What do you think?

Reply:
I absolutely agree that this needs unification towards
> [...] there is only one method called Optional<Day> getValue() with the exact behavior of getDayOrThrowIfInvalid()

Regarding the name of that method: `parseValue` > `getValue`, for the stated reason.

Regarding the backlog entry: Yes, should have that.

Reply:
ok, I renamed the 3 methods in `DomDayInput`, `DomTimeInput` and `DomDayTimeInput` to `parseValue`